File: Programmers.co.kr/Level 2/124 나라의 숫자/solution.py
def solution(n):
    answer = ''
    
    q = (n - 1) // 3
    r = (n - 1) % 3
    
    if q > 0:
        m = solution(q)
        answer = answer + m
    
    answer = answer + str(2**(r))
        
    return answer

File: Programmers.co.kr/Level 2/124 나라의 숫자/solution.cpp
#include <math.h>
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "";
    
    int q = (n - 1)/3;
    int r = (n - 1) - 3*q;
    
    if(q > 0)
    {
        answer = answer + solution(q);
    }
    
    answer = answer + std::to_string(int(pow(2, r)));
    
    return answer;
}
